app/components/govuk_component/pagination_component.rb

```diff
@@ -0,0 +1,191 @@
+class GovukComponent::PaginationComponent < GovukComponent::Base
+  include Pagy::UrlHelpers
+
+  attr_reader :pagy
+
+  renders_one :previous_item, ->(href:, text: "Previous", label: nil, classes: [], html_attributes: {}) do
+    PreviousItem.new(href: href, text: text, label: label, block_mode: block_mode?, classes: classes, html_attributes: html_attributes)
+  end
+
+  renders_one :next_item, ->(href:, text: "Next", label: nil, classes: [], html_attributes: {}) do
+    NextItem.new(href: href, text: text, label: label, block_mode: block_mode?, classes: classes, html_attributes: html_attributes)
+  end
```

Comment:
We need to use a [lambda slot](https://viewcomponent.org/guide/slots.html#lambda-slots) here so we can pass the `block_mode?` down to the children.

It's only _really_ needed on the `NextItem`, because in block mode the arrow goes before the title (`→ Next` instead of `Next →`), but applying it to both at least makes them match 😅